cpp/src/structure/create_graph_from_edgelist_impl.cuh

```diff
@@ -530,59 +619,110 @@ create_graph_from_edgelist_impl(
       (*edge_partition_edgelist_weights)[i].resize(0, handle.get_stream());
       (*edge_partition_edgelist_weights)[i].shrink_to_fit(handle.get_stream());
     }
-    if (edge_partition_edgelist_id_type_pairs) {
-      std::get<0>((*edge_partition_edgelist_id_type_pairs)[i]).resize(0, handle.get_stream());
-      std::get<0>((*edge_partition_edgelist_id_type_pairs)[i]).shrink_to_fit(handle.get_stream());
-      std::get<1>((*edge_partition_edgelist_id_type_pairs)[i]).resize(0, handle.get_stream());
-      std::get<1>((*edge_partition_edgelist_id_type_pairs)[i]).shrink_to_fit(handle.get_stream());
+    if (edge_partition_edgelist_edge_ids) {
+      (*edge_partition_edgelist_edge_ids)[i].resize(0, handle.get_stream());
+      (*edge_partition_edgelist_edge_ids)[i].shrink_to_fit(handle.get_stream());
+    }
+    if (edge_partition_edgelist_edge_types) {
+      (*edge_partition_edgelist_edge_types)[i].resize(0, handle.get_stream());
+      (*edge_partition_edgelist_edge_types)[i].shrink_to_fit(handle.get_stream());
+    }
+    if (edge_partition_dcs_nzd_vertices) {
+      (*edge_partition_dcs_nzd_vertices).push_back(std::move(*dcs_nzd_vertices));
     }
 
     edge_partition_offsets.push_back(std::move(offsets));
     edge_partition_indices.push_back(std::move(indices));
     if (edge_partition_weights) { (*edge_partition_weights).push_back(std::move(*weights)); }
-    if (edge_partition_id_type_pairs) {
-      (*edge_partition_id_type_pairs).push_back(std::move(*id_type_pairs));
-    }
-    if (edge_partition_dcs_nzd_vertices) {
-      (*edge_partition_dcs_nzd_vertices).push_back(std::move(*dcs_nzd_vertices));
```

Comment:
Why did you delete this?

Reply:
I didn't delete this.  It's up higher (line 630 on the right).  Not sure why I needed up moving it before the others instead of leaving it here.

Reply:
Could you move the code back here? It make the code easy to follow.